provider/identityprovider/htpasswd.go

```diff
@@ -75,19 +76,23 @@ var htpasswdUserList = map[string]schema.Attribute{
 	},
 }
 
-func CreateHTPasswdIDPBuilder(ctx context.Context, state *HTPasswdIdentityProvider) *cmv1.HTPasswdIdentityProviderBuilder {
+func CreateHTPasswdIDPBuilder(ctx context.Context, state *HTPasswdIdentityProvider) (*cmv1.HTPasswdIdentityProviderBuilder, error) {
 	builder := cmv1.NewHTPasswdIdentityProvider()
 	userListBuilder := cmv1.NewHTPasswdUserList()
 	userList := []*cmv1.HTPasswdUserBuilder{}
 	for _, user := range state.Users {
+		hashedPwd, err := idputils.GenerateHTPasswdCompatibleHash(user.Password.ValueString())
+		if err != nil {
+			return nil, err
+		}
 		userBuilder := &cmv1.HTPasswdUserBuilder{}
 		userBuilder.Username(user.Username.ValueString())
-		userBuilder.Password(user.Password.ValueString())
+		userBuilder.HashedPassword(hashedPwd)
```

Comment:
We should keep the option for users to add their own password, no?

Reply:
yes, the hashedpassword is calculated from the user provided one: 
hashedPwd, err := idputils.GenerateHTPasswdCompatibleHash(user.Password.ValueString())

for idp creation, we don't provide the option to generate the password. Just wants to keep the same behavior here as rosa cli